### PR TITLE
Account page

### PR DESCRIPTION
**Pull request by alonpeled44**

### CODE REVIEW

File: components/account/AccountUserName.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+import styles from "@/styles/components/account/account-user-name.module.scss";
+
+export default function AccountUserName() {
+	const [isEditActive, setIsEditActive] = useState(false);
+	const [userName, setUserName] = useState("Lio Giladi");
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:6]: There's no purpose for this state other than initializing the latter. remove it.

Comment (liogiladi) [inline components/account/AccountUserName.jsx:6]: where is "Lio Giladi" coming from? The default value should come from external source -> should come from a prop

File: components/account/AccountUserName.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+				<input
+					type='text'
+					value={changingUserName}
+					onChange={onUserNameChange}
+					className={styles[`user-name-input-${isEditActive ? "save" : "edit"}`]}
+					disabled={!isEditActive}
+				/>
+			)}
+
+			<button className={styles.filled} onClick={onEditbuttonClick}>
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:39]: you've made a component that has the same styles (with minor differences), use it instead

File: components/account/AccountUserName.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			{!isEditActive ? (
+				<span className={styles["user-name-view"]} title={changingUserName}>
+					{changingUserName}
+				</span>
+			) : (
+				<input
+					type='text'
+					value={changingUserName}
+					onChange={onUserNameChange}
+					className={styles[`user-name-input-${isEditActive ? "save" : "edit"}`]}
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:34]: why do you use 2 styles? the input renders only when isEditActive is true, so the class name will always be "user-name-input-save".
overall there's no reason to give this input a classname

File: components/account/AccountUserName.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+					disabled={!isEditActive}
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:35]: isn't relevant anymore

File: components/account/AccountUserName.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	};
+
+	const onUserNameChange = (event) => {
+		setChangingUsername(event.target.value);
+	};
+
+	return (
+		<section className={styles["account-user-name"]}>
+			{!isEditActive ? (
+				<span className={styles["user-name-view"]} title={changingUserName}>
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:26]: redundant use of class name

File: components/account/AccountUserName.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		setIsEditActive((prev) => {
+			if (prev === true) {
+				setUserName(changingUserName);
+			}
+
+			return !prev;
+		});
+	};
+
+	const onUserNameChange = (event) => {
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:19]: I know it's your style to declare a function for every event, but it _would_ be cleaner to just:
```jsx
<input otherProps onChange={(e) => setUserName(e.target.value)} />
```

File: components/account/AccountUserName.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import styles from "@/styles/components/account/account-user-name.module.scss";
+
+export default function AccountUserName() {
+	const [isEditActive, setIsEditActive] = useState(false);
+	const [userName, setUserName] = useState("Lio Giladi");
+	const [changingUserName, setChangingUsername] = useState(userName);
+
+	const onEditbuttonClick = () => {
+		setIsEditActive((prev) => {
+			if (prev === true) {
```
Comment (liogiladi) [inline components/account/AccountUserName.jsx:11]:
```js
if (prev === true)
```

is the same as:
```js
if(prev)
```

File: pages/account.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+		} catch {}
+	};
+
+	return (
+		<main id={styles.main}>
+			<label for='profilePicInput' className={styles["profile-pic-input"]}>
+				<img src={profilePicSrc} alt="user's photo" />
+			</label>
+			<AccountUserName />
+			<input type='file' id='profilePicInput' onChange={onUpload} accept='image/png, image/jpg, image/jpeg' />
```
Comment (liogiladi) [inline pages/account.jsx:22]: the input should be inside of the label or at least be next to it below. The dividing Component makes things unclear.

File: pages/account.jsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+import styles from "@/styles/pages/account.module.scss";
+import AccountUserName from "@/components/account/AccountUserName";
+
+export default function Account() {
+	const daysAlive = 9;
+	const [profilePicSrc, setProfilePicSrc] = useState("/lio_img.png");
+
+	const onUpload = (event) => {
+		try {
```
Comment (liogiladi) [inline pages/account.jsx:10]: why the try catch?

File: styles/components/account/account-user-name.module.scss [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+		height: 55px;
+		position: relative;
+		background-color: transparent;
+		text-align: center;
+		font-size: 3.5rem;
+		font-weight: 500;
+		color: var(--secondary-color);
+	}
+
+	.user-name-input-save {
```
Comment (liogiladi) [inline styles/components/account/account-user-name.module.scss:19]: described it in another comment but there's no need for this class name and the one below

File: styles/components/account/account-user-name.module.scss [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+		border: none;
+	}
+
+	.user-name-view {
+		line-height: 3.5rem;
+		overflow: hidden;
+		text-overflow: ellipsis;
+		text-wrap: nowrap;
+		display: inline-block;
+		padding-bottom: 3px;
```
Comment (liogiladi) [inline styles/components/account/account-user-name.module.scss:34]: no need for the padding-bottom

File: styles/components/account/account-user-name.module.scss [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+		font-family: Kanit;
+		width: 275px;
+		position: relative;
```
Comment (liogiladi) [inline styles/components/account/account-user-name.module.scss:37]: this does nothing

File: styles/components/account/account-user-name.module.scss [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+		text-align: center;
+		font-size: 3.5rem;
+		font-weight: 500;
+		color: var(--secondary-color);
+	}
+
+	button {
```
Comment (liogiladi) [inline styles/components/account/account-user-name.module.scss:44]: described in earlier comment but use the component you already made

File: styles/components/account/account-user-name.module.scss [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+		gap: 25px;
+
+		input[type="text"] {
+			width: 180px;
+			height: 45px;
+			font-size: 2.5rem;
+			padding-bottom: 0px;
+		}
+
+		.user-name-view {
```
Comment (liogiladi) [inline styles/components/account/account-user-name.module.scss:78]: repeating declaration with the input put in a chaining selector

**Review by liogiladi — CHANGES_REQUESTED** [review]
little bit

File: pages/account.jsx (hunk 8 above)
Comment (alonpeled44) [reply pages/account.jsx]: noted

File: components/account/AccountUserName.jsx (hunk 7 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: noted

File: components/account/AccountUserName.jsx (hunk 1 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: noted

Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: noted

File: components/account/AccountUserName.jsx (hunk 3 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx:27]: noted

File: components/account/AccountUserName.jsx (hunk 5 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: noted

File: styles/components/account/account-user-name.module.scss (hunk 10 above)
Comment (alonpeled44) [reply styles/components/account/account-user-name.module.scss:19]: noted

File: styles/components/account/account-user-name.module.scss (hunk 11 above)
Comment (alonpeled44) [reply styles/components/account/account-user-name.module.scss:34]: noted

File: styles/components/account/account-user-name.module.scss (hunk 12 above)
Comment (alonpeled44) [reply styles/components/account/account-user-name.module.scss:37]: noted

File: styles/components/account/account-user-name.module.scss (hunk 14 above)
Comment (alonpeled44) [reply styles/components/account/account-user-name.module.scss:78]: noted

File: components/account/AccountUserName.jsx (hunk 4 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: noted

File: components/account/AccountUserName.jsx (hunk 6 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: noted

File: styles/components/account/account-user-name.module.scss (hunk 13 above)
Comment (alonpeled44) [reply styles/components/account/account-user-name.module.scss:30]: nuh-uh

File: components/account/AccountUserName.jsx (hunk 2 above)
Comment (alonpeled44) [reply components/account/AccountUserName.jsx]: nuh-uh

**Review by liogiladi — APPROVED** [review]
* change the for to htmlFor
* update old dependencies

**Pull request merged**